### PR TITLE
Remove Akka-Enhancements relevant docs

### PR DESCRIPTION
**Pull request by dnychennnn**
- Discussion: https://lists.apache.org/thread/8orjtlx2w42y3j45pswmf9nbtf64gp77
- Issue: #122

### CODE REVIEW

File: docs/src/main/paradox/general/configuration-reference.md [hunk 1]
```diff
@@ -3,8 +3,7 @@
 Each Pekko module has a `reference.conf` file with the default values.
 
 Make your edits/overrides in your `application.conf`. Don't override default values if
-you are not sure of the implications. [Pekko Config Checker](https://doc.akka.io/docs/akka-enhancements/current/config-checker.html)
-is a useful tool for finding potential configuration issues.
+you are not sure of the implications. Pekko Config Checker is a useful tool for finding potential configuration issues.
```
Comment (pjfanning) [inline docs/src/main/paradox/general/configuration-reference.md:6]: There is no such thing as a `Pekko Config Checker`

File: docs/src/main/paradox/common/other-modules.md [hunk 2]
```diff
@@ -52,11 +52,11 @@ Use [Google Cloud Spanner](https://cloud.google.com/spanner/) as Pekko Persisten
...
-* [Pekko Configuration Checker](https://doc.akka.io/docs/akka-enhancements/current/config-checker.html)
-* [Pekko Diagnostics Recorder](https://doc.akka.io/docs/akka-enhancements/current/diagnostics-recorder.html)
+* Pekko Thread Starvation Detector
+* Pekko Configuration Checker
+* Pekko Diagnostics Recorder
 
 ## Pekko Persistence Enhancements
 
-* [Pekko GDPR for Persistence](https://doc.akka.io/docs/akka-enhancements/current/gdpr/index.html)
+* Pekko GDPR for Persistence
```
Comment (pjfanning) [inline docs/src/main/paradox/common/other-modules.md:61]: there is no point keeping section - there is no such thing as `Pekko GDPR for Persistence` - likewise for all the `Pekko Resilience Enhancements`

File: docs/src/main/paradox/general/configuration-reference.md (hunk 1 above)
Comment (dnychennnn) [reply docs/src/main/paradox/general/configuration-reference.md:6]: Understood, let me remove the line.

File: docs/src/main/paradox/common/other-modules.md (hunk 2 above)
Comment (dnychennnn) [reply docs/src/main/paradox/common/other-modules.md:61]: removed

**Review by pjfanning — APPROVED** [review]
lgtm - thanks

**Pull request merged**